### PR TITLE
feat: Mithril snapshot reader from Point

### PR DESCRIPTION
**Pull request by Mr-Leshiy**
Added new functionality for the `pallas-hardano` crate:
- `read_blocks_from_point(dir: &Path, point: Point)` - Creates a block's iterator starting from the provided `point`. Performs a binary search over the chunks to find a **potential** chunk that could contain the provided `point`. Once the **potential** chunk is found, it iterates over it until it reaches the specified `point` and returns an iterator starting from this `point`. If the `point` is not found within the chunk, an error is returned.
-  `get_tip(dir: &Path)` - Returns a `Point` value for the latest block. It iterates to the end of the last chunk and returns the `Point` value of the block.

### CODE REVIEW

**Comment by Mr-Leshiy** [comment]
DONT MERGE THIS PR TO THIS REPO

related to this issue https://github.com/input-output-hk/hermes/issues/42

**Comment by Mr-Leshiy** [comment]
related to this issue https://github.com/input-output-hk/hermes/issues/42

**Review by stevenj — CHANGES_REQUESTED** [review]
Looks good,
Just the few things we discussed.
Returning an error when attempting to read origin and it doesn't;t exist in the data.
Having an error type for the errors we produce internally, and documenting in the public API functions all the possible error types the function can return.

File: pallas-hardano/src/storage/immutable/mod.rs [hunk 1]
```diff
@@ -59,13 +150,230 @@ pub fn read_blocks(dir: &Path) -> Result<impl Iterator<Item = FallibleBlock>, st
...
+                .map_while(Result::ok)
+                .flatten()
+                .peekable();
+
+            // check the first block
+            match iter.peek() {
+                Some(Ok(block_data)) => {
+                    let block = MultiEraBlock::decode(&block_data)?;
+                    // check that the first block is genesis
+                    if block.slot() == 0 && block.number() == 0 {
```
Comment (bkioshn) [inline pallas-hardano/src/storage/immutable/mod.rs:216]: Why do we need to check for `slot`?  since it is genesis

Comment (Mr-Leshiy) [reply pallas-hardano/src/storage/immutable/mod.rs:216]: For the genesis block `slot` equals to `0` same as for the `number`.
Generally it should be enough to check only for `number` field, but as this check does not cost too much for as and for additional validation I think it is better to check both these fields.

**Review by stevenj — APPROVED** [review]
LGTM

**Pull request closed without merging**